JSTutorial/hw_7/task_2.js

```diff
@@ -0,0 +1,299 @@
+ *
+ * Пример:
+ * deleteDepartment(3)
+ *
+ *
+ * 9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+ *
+ * Пример:
+ * moveEmployees(2, 3)
+ */
+const enterprises = [
+    {
+        id: 1,
+        name: "Предприятие 1",
+        departments: [
+            {
+                id: 2,
+                name: "Отдел тестирования",
+                employees_count: 10,
+            },
+            {
+                id: 3,
+                name: "Отдел маркетинга",
+                employees_count: 20,
+            },
+            {
+                id: 4,
+                name: "Администрация",
+                employees_count: 15,
+            },
+        ]
+    },
+    {
+        id: 5,
+        name: "Предприятие 2",
+        departments: [
+            {
+                id: 6,
+                name: "Отдел разработки",
+                employees_count: 50,
+            },
+            {
+                id: 7,
+                name: "Отдел маркетинга",
+                employees_count: 20,
+            },
+            {
+                id: 8,
+                name: "Отдел охраны труда",
+                employees_count: 5,
+            },
+        ]
+    },
+    {
+        id: 9,
+        name: "Предприятие 3",
+        departments: [
+            {
+                id: 10,
+                name: "Отдел аналитики",
+                employees_count: 0,
+            },
+        ]
+    }
+]
+
+//1
+function countEmployees(departments) {
+    return departments.reduce(function(total, department) {
+        return total + department.employees_count;
+    }, 0);
+}
+console.log(countEmployees(enterprises[0].departments));
+const getCommonInfo = (company) => {
+    company.forEach(enterprise => {
+            console.log(`${enterprise.name} (${countEmployees(enterprise.departments)} сотрудников)`);
+            enterprise.departments.forEach(department => console.log(`- ${department.name} (${department.employees_count})`));
+    })
+}
+getCommonInfo(enterprises);
+
+//2
+function getFactoryByNameORId(key) {
+    let name = null;
+    enterprises.forEach(enterprise => {
+        enterprise.departments.forEach(department => {
+            if (department.id === key || department.name === key) name = enterprise.name;
+        })
+    })
+    return name;
+}
+console.log(getFactoryByNameORId("Отдел тестирования"))
+console.log(getFactoryByNameORId(6))
+
+//3
+function addEnterprise(name) {
+    const nejObj = {
+        name,
+        departments: [],
+        id : enterprises.length - 1,
+    }
+    enterprises.push(nejObj);
+}
+
+addEnterprise("Предприятие 4")
+console.log(enterprises);
+
+//4
+function getEnterpriseByIdOrName(key) {
```

Comment:
Try specifying a search array as a function parameter so that it works only with the data that was passed, making it independent
```suggestion
function getEnterpriseByIdOrName(key, arr) {
```